src/deploy/scope_subscription.ts

```diff
@@ -37,6 +36,8 @@ export async function DeploySubscriptionScope(azPath: string, location: string,
     // validate the deployment
     info("Validating template...")
     await exec(`"${azPath}" deployment sub validate ${azDeployParameters} -o json`, [], { silent: true, failOnStdErr: true });
```

Comment:
We have seen cases where even though validation fails, the template can still get deployed successfully. 
Hence if validation fails and if `validationOnly` is false, then action should not fail and just show a warning and go ahead with deployment. If `validationOnly` is true then obviously it should fail the action.

Reply:
Also `failOnStdErr` is tricky here as it will cause failure and hence such failure will need to be caughts and handled properly.